### PR TITLE
Fix Run Query for widgets

### PR DESCRIPTION
**Pull request by abist**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/7387

The issue was that we weren't providing any query string to run the query with.

### CODE REVIEW

File: src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts [hunk 1]
```diff
@@ -23,6 +22,14 @@ export class RunInsightQueryAction extends Action {
 	}
 
 	public run(context: InsightActionContext): Promise<boolean> {
-		return this.instantiationService.invokeFunction(openNewQuery, context.profile, undefined, RunQueryOnConnectionMode.executeQuery).then(() => true, () => false);
+		let queryString: string = undefined;
+		if (context.insight && context.insight.query) {
+			if (typeof (context.insight.query) === 'string') {
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: should just use the typing helper functions

File: src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts [hunk 2]
```diff
@@ -23,6 +22,14 @@ export class RunInsightQueryAction extends Action {
 	}
 
 	public run(context: InsightActionContext): Promise<boolean> {
-		return this.instantiationService.invokeFunction(openNewQuery, context.profile, undefined, RunQueryOnConnectionMode.executeQuery).then(() => true, () => false);
+		let queryString: string = undefined;
+		if (context.insight && context.insight.query) {
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: Should we just return if there is no query to open?

File: src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts [hunk 3]
```diff
@@ -28,7 +27,7 @@ export class RunInsightQueryAction extends Action {
 			if (typeof (context.insight.query) === 'string') {
 				queryString = context.insight.query;
 			} else {
-				queryString = context.insight.query.join(os.EOL);
+				queryString = context.insight.query.join('\n');
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: You should get the newline from the TextResourcePropertiesService https://github.com/microsoft/azuredatastudio/blob/eng/linux/src/vs/workbench/services/textresourceProperties/common/textResourcePropertiesService.ts#L20

File: src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts (hunk 1 above)
Comment (abist) [reply src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: where are the typing helper functions?

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/types.ts#L32

File: src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts [hunk 4]
```diff
@@ -9,27 +9,34 @@ import { RunQueryOnConnectionMode } from 'sql/platform/connection/common/connect
...
 		if (context.insight && context.insight.query) {
-			if (typeof (context.insight.query) === 'string') {
+			if (isString(context.insight.query)) {
 				queryString = context.insight.query;
 			} else {
-				queryString = context.insight.query.join('\n');
+				queryString = context.insight.query.join(eol);
 			}
+		} else {
+			return undefined;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts]: Just return false here

**Pull request merged**